### PR TITLE
programab session focus wake and sleep

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/ProgramAB.java [hunk 1]
```diff
@@ -1329,7 +1342,8 @@ public void onUtterance(Utterance utterance) throws Exception {
         // TODO: don't talk to bots.. it won't go well..
         // TODO: the discord api can provide use the list of mentioned users.
         // for now.. we'll just see if we see Mr. Turing as a substring.
-        if (utterance.text.contains(botName)) {
+        sleep = (sleep || utterance.text.contains("@")) && !utterance.text.contains(botName);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/ProgramAB.java:1369]: i'd like to preserve existing behavior as default and make this behavior a configuration option exposed in the config

File: src/main/java/org/myrobotlab/service/ProgramAB.java [hunk 2]
```diff
@@ -1296,6 +1298,16 @@ synchronized public void addCategoryToFile(Bot bot, Category c) {
       error(e);
     }
   }
+  
+  boolean sleep = false;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/ProgramAB.java]: member variables should go to the top of the class.

File: src/main/java/org/myrobotlab/service/ProgramAB.java [hunk 3]
```diff
@@ -1296,6 +1298,16 @@ synchronized public void addCategoryToFile(Bot bot, Category c) {
       error(e);
     }
   }
+  
+  boolean sleep = false;
+  
+  public void wake() {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/ProgramAB.java:1325]: are you intending these to be invoked via aiml ?

File: src/main/java/org/myrobotlab/service/ProgramAB.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/ProgramAB.java]: agreed fixed

File: src/main/java/org/myrobotlab/service/ProgramAB.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/ProgramAB.java:1369]: agreed fixed

File: src/main/java/org/myrobotlab/service/ProgramAB.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/ProgramAB.java:1325]: yes - it worky ... tried first with single method and boolean ... and that failed ...   :(

File: src/main/java/org/myrobotlab/service/config/ProgramABConfig.java [hunk 4]
```diff
@@ -15,5 +15,15 @@ public class ProgramABConfig extends ServiceConfig {
   // search engine - currently wikipedia
   // search for sraix
   public String search;
+  
+  /**
+   * enables global session and sleep/wake functionality
+   */
+  public boolean useGlobalSession;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/config/ProgramABConfig.java]: you've added this configuration variable, but it isn't actually used anywhere. please remove.

File: pom.xml [hunk 5]
```diff
@@ -349,23 +349,11 @@
 <!-- DocumentPipeline end -->
 
 <!-- Elasticsearch begin -->
-  <dependency>
```
Comment (kwatters) [inline pom.xml]: Why are there pom changes in this PR?  please revert pom.xml

Comment (supertick) [reply pom.xml]: pretty sure the current was out of spec not being generated ... but whatever, i reverted

**Pull request merged**